url2df0f7c811_przemekgalazka_legacycode_tgz-pJ8-pl_cc_core_cmd_CommandJ8/src/pl/cc/core/cmd/Command.java

```diff
@@ -0,0 +1,474 @@
+  public static final int CMD_EVENT_QUEUE_LEAVE = 11;
+
+  public static final int CMD_EVENT_AGENT_ADD_TO_QUEUE = 21;
+  public static final int CMD_EVENT_AGENT_REMOVE_FROM_QUEUE = 22;
+  public static final int CMD_EVENT_AGENT_NEW_CALL = 23;
+  public static final int CMD_EVENT_AGENT_HANGUP = 24;
+  public static final int CMD_EVENT_AGENT_CONNECT = 25;
+  public static final int CMD_EVENT_AGENT_PAUSE = 26;
+  public static final int CMD_EVENT_AGENT_UNPAUSE = 27;
+  public static final int CMD_EVENT_AGENT_LOGIN = 28;
+  public static final int CMD_EVENT_AGENT_LOGOUT = 29;
+  public static final int CMD_EVENT_AGENT_QUEUE_MEMBER = 30;
+  public static final int CMD_EVENT_AGENT_COMPLETED = 31;
+  public static final int CMD_EVENT_AGENT_CALL_TIME_BAD = 32;
+
+  public static final int CMD_WELCOME = 101;
+  public static final int CMD_PONG = 102;
+  public static final int CMD_OK_LOGGING_OK = 103;
+  public static final int CMD_OK_LOGGING_OK_SUPERVISOR = 104;
+  public static final int CMD_OK_LOGGING_ERROR = 105;
+  public static final int CMD_OK_LOGGED_OUT = 106;
+
+  public static final int CMD_QUEUE_ACTION_ADD = 201;
+  public static final int CMD_QUEUE_ACTION_REMOVE = 202;
+
+  public static final int CMD_OK_WELCOME = 301;
+  public static final int CMD_OK_PAUSED = 302;
+
+  public static final int CMD_ERR_CONNECTION_REPLACED = 401;
+
+  public static final int CMD_EVENT_CALL_TAG = 501;
+  public static final int CMD_EVENT_PROXYCLIENT_CALL_TAG = 502;
+
+
+  /**
+   * Zdarzenia płynące od CCProxyClient -> CCProxy
+   */
+  public static final int CMD_EVENT_PROXYCLIENT_PAUSE = 501;
+
+  enum CmdType {INFO, EVENT, UNKNOWN}
+
+  ;
+
+  //static Matcher m;
+  static Pattern p = Pattern.compile(".*");
+
+  public static final String REGEX_CHARACTERS_INFO = "\\w\\s\\.,/:\\-ęóąśłżźćńĘÓĄŚŁŻŹĆŃ<>\\{\\}\\|\\(\\)";
+  public static final String REGEX_CHARACTERS_EVENT = "^\\[\\]";
+
+
+  public abstract int getType();
+
+
+  private String orginalLine;
+  private String connectionId;
+
+
+  public Command(String orginalLine, String connectionId) {
+    this.orginalLine = orginalLine;
+    this.connectionId = connectionId;
+  }
+
+
+  protected Command(String orginalLine) {
+    this.orginalLine = orginalLine;
+  }
+
+
+  protected static boolean equalsExtended(String line, String searching) {
+    if (line == null) return false;
+    String l2 = line.replaceAll("\\s", "").replaceAll("\\.", "");
+    String s2 = searching.replaceAll("\\s", "").replaceAll("\\.", "");
+    return l2.toLowerCase().equals(s2.toLowerCase());
+  }
+
+
+  private static String getValueINFO(String line, String name) {
+    synchronized (p) {
+      //+INFO Queue 'rekawek', Count '0'
+      Matcher m = Pattern.compile(".*" + name + "[\\s,:]*'([" + REGEX_CHARACTERS_INFO + "]+)'.*", Pattern.CASE_INSENSITIVE).matcher(line);
+      if ((m.find()) && (m.groupCount() == 1)) {
+        return m.group(1);
+      } else {
+        return null;
+      }
+    }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 105:113, 118:126.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.